Add newsfeeds base to master branch

bmack123 opened the pull request.

bmack123 committed:
Include necessary gems for testing purposes

bmack123 committed:
Newsfeed model creation and specs

bmack123 committed:
Newsfeeds Controller creation and specs

bmack123 committed:
Seed database with newsfeeds

bmack123 committed:
Install packages for routing and requests in javascript

bmack123 committed:
Implement newsfeed items display as a list and individual

AJMiller commented on app/controllers/api/newsfeeds_controller.rb:
```diff
@@ -0,0 +1,14 @@
+
+class Api::NewsfeedsController < ApplicationController
+  def index
+    render json: Newsfeed.all.to_json
```
Are these `.to_json`'s necessary?

AJMiller replied:
nope removed

AJMiller commented on app/controllers/api/newsfeeds_controller.rb:
```diff
@@ -0,0 +1,14 @@
+    newsfeed = Newsfeed.find(params[:id].to_i)
+    render json: newsfeed.to_json
+  rescue ActiveRecord::RecordNotFound
+    render json: nil, status: :not_found
```
I believe `json: nil,` is unnecessary here

AJMiller replied:
it is, but we have a test for an empty json body so I left it

bmack123 replied:
otherwise it tries to render a template by default and doesn't find it. I tried `head` too but it wouldn't let me do `head status: :not_found`. But I might just not know the magic words.

bmack123 replied:
`head :not_found` were the magic words

bmack123 replied:
rebased back into d5fd6fdea8e7158dc0ec7566b5e2d0c7ba1a9bd2

AJMiller commented on app/helpers/api/newsfeeds_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module Api::NewsfeedsHelper
```
not sure how useful this is to have?

bmack123 replied:
Probably not at all, I wasn't sure if we wanted to remove things in case they add to this in the future (it came from the generator)

bmack123 replied:
removed files in 3c1a6aaba8693d0baaf5d86b95bf996bfd3ca4f3

AJMiller commented on app/assets/javascripts/api/newsfeeds.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Do we need this?

bmack123 replied:
No, and I think that we live in a world where they definitely won't need i even as we expand so I think we can safely remove it.

bmack123 replied:
removed files in 3c1a6aaba8693d0baaf5d86b95bf996bfd3ca4f3

AJMiller commented on client/src/App.js:
```diff
@@ -1,34 +1,19 @@
-      </header>
+    <div className='App'>
+      <Router>
+        <Route exact path='/' component={NewsfeedList} />
```
Do we need to do `<Route exact` here or can we just do `<IndexRoute` ?

bmack123 replied:
Probably, I was just following what they did on their site.

bmack123 replied:
We actually can't do `IndexRoute` its gone in the latest version of `react-router` (the last 2 versions actually)

AJMiller replied:
looks like the new react-router does things this way

AJMiller commented on client/src/App.js:
```diff
@@ -1,34 +1,19 @@
+    <div className='App'>
+      <Router>
+        <Route exact path='/' component={NewsfeedList} />
+        <Route path='/newsfeeds/:id' component={NewsfeedItem} />
```
Maybe we start this as `/:id` and open the door to discussing route structure improvements?

bmack123 replied:
:+1:

bmack123 replied:
rebased back into 7125dacd272f1daf2111e74f49e18e913c141ecd

AJMiller commented on client/yarn.lock:
```diff
@@ -859,6 +859,13 @@
   dependencies:
```
Not sure we need to make yarn a requirement or just use npm for this repo?

AJMiller replied:
swapped out yarn.lock for package-lock.json so we don't need yarn for this repo

bmack123 replied:
we don't

AJMiller commented on db/migrate/20190808134440_create_newsfeeds.rb:
```diff
@@ -0,0 +1,10 @@
+class CreateNewsfeeds < ActiveRecord::Migration[5.2]
+  def change
+    create_table :newsfeeds do |t|
+      t.string :title
```
we should probably add database validations here
```suggestion
      t.string :title, null: false
```

bmack123 replied:
rebased into 767a80bca6c968c97783fbce30aa180b4dcacfb7

AJMiller commented on spec/controllers/api/newsfeeds_controller_spec.rb:
```diff
@@ -0,0 +1,74 @@
+
+    before do
+      FactoryBot.create_list(:newsfeed, number_of_items)
+      subject
```
`subject` shouldn't be called in a before block

AJMiller replied:
did it

bmack123 replied:
I believe that you took care of this good sir

Commit:
Removed title length validation

It can cause random test failures if the factory happens to make a title longer than 25 chars. Didn't see super valuable to keep.

Commit:
Remove unnecessary complications from newsfeed controller

Commit:
Make controller specs more ruby-esque

fixup controller spec

Commit:
swap yarn.lock for package-lock.json

Candidate shouldn't need to install yarn to get up and running.

bmack123 committed:
Remove unnecessary newsfeed generated files

Commit:
Update README and SETUP for new story

bmack123 reviewed (commented):
LGTM

derosm2 reviewed (approved):
👍
